docs/flinkDev/building.md

```diff
@@ -105,20 +105,25 @@ Flink [shades away](https://maven.apache.org/plugins/maven-shade-plugin/) some o
 
 The dependency shading mechanism was recently changed in Maven and requires users to build Flink slightly differently, depending on their Maven version:
 
-**Maven 3.1.x and 3.2.x**
+**Maven and 3.2.x**
 It is sufficient to call `mvn clean install -DskipTests` in the root directory of Flink code base.
 
 **Maven 3.3.x**
-The build has to be done in two steps: First in the base directory, then in the distribution project:
+The build has to be done in two steps: First in the base directory, then in shaded modules, such as the distribution and the filesystems:
 
 {% highlight bash %}
 mvn clean install -DskipTests
 cd flink-dist
 mvn clean install
+cd ../flink-filesystems/flink-s3-fs-presto/
```

Comment:
This should be done before flink-dist (otherwise it won't be picked up by flink-dist).

Furthermore, there are so many instances by now (filesystems, metric reporters, sql connectors) that we simply should not list maven 3.3 as an option.

Reply:
ah, damn, you are right.